File: LeetCode/53MaximumSubarray/MaximumSubarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int ans = nums[0];
        int currSum = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (currSum < 0) {
                currSum = 0;
            }
            currSum += nums[i];
            if (currSum > ans) {
                ans = currSum;
            }
        }
        return ans;
    }
};

File: LeetCode/53MaximumSubarray/MaximumSubarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        # Single pass -> Time:O(n), Space:O(1)
        ans = nums[0]
        curr_sum = 0
        for i in nums:
            if curr_sum < 0:
                curr_sum = 0
            curr_sum += i
            ans = max(ans, curr_sum)
        return ans
